### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by Devjacky**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -18,4 +26,15 @@ export class AuthController {
     this.logger.log(message);
     return this.authService.signup(createUserDto);
   }
+
+  @UseGuards(LocalAuthGuard)
+  @Post('login')
+  login(@Request() req) {
```
Comment (kzkaneoka) [inline src/auth/auth.controller.ts:32]: Let's add `Promise<any>` as a return type!

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -17,4 +19,26 @@ export class AuthService {
...
+    this.logger.log(message);
+    const user = await this.usersService.findByUsername(username);
+    if (user && user.password === password) {
+      const { password, ...result } = user;
+      return result;
+    }
+    return null;
+  }
+
+  // user is object. any better types?
```
Comment (kzkaneoka) [inline src/auth/auth.service.ts:35]: Maybe, `Promise<any>`?

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -17,4 +19,26 @@ export class AuthService {
...
+  async login(user: any) {
```
Comment (kzkaneoka) [inline src/auth/auth.service.ts:36]: I believe you can use `UserModel` instead of `any` here

File: src/auth/local.strategy.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { AuthService } from './auth.service';
+
+@Injectable()
+export class LocalStrategy extends PassportStrategy(Strategy) {
+  constructor(private authService: AuthService) {
+    super();
+  }
+
+  async validate(username: string, password: string): Promise<any> {
```
Comment (kzkaneoka) [inline src/auth/local.strategy.ts:12]: Let's use `Promise<UserModel>` instead of  `Promise<any>` here!

**Review by kzkaneoka — COMMENTED** [review]
I've reviewed and commented. Please check them. Thanks!

**Comment by Devjacky** [comment]
I have made the necessary changes based on your comments. Please let me know if there's anything else I'm missing. Thank you!

**Comment by kzkaneoka** [comment]
LGTM!

**Pull request merged**